src/app/lib/views/torrent_collection.js

```diff
@@ -76,179 +107,193 @@
-                                    peers: ldata.leechers,
-                                    size: ldata.size,
+                                    title: item.title,
+                                    magnet: item.download,
+                                    seeds: item.seeders,
+                                    peers: item.leechers,
+                                    size: Common.fileSize(parseInt(item.size)),
                                     index: index
                                 };
-                                indx++;
+
                                 if (item.title.match(/trailer/i) !== null && input.match(/trailer/i) === null) {
                                     return;
                                 }
-                                results.push(itemModel);
+                                results1.push(itemModel);
                                 index++;
-                            }).catch(function (err) {
-                                throw 'OMG info failed';
                             });
-                        });
-                    }).catch(function (err) {
-                        console.error('OMG search:', err);
-                        resolve(results);
-                    });
-                });
-            };
-
-            var piratebay = function () {
-                return new Promise(function (resolve) {
-                    var results = [];
-                    setTimeout(function () {
-                        resolve(results);
-                    }, 6000);
-                    var tpb = torrentCollection.tpb;
-                    tpb.search(input.toLocaleLowerCase(), {
-                        category: 'video',
-                        page : 0,
-                        orderBy: 'seeds',
-                        sortBy: 'desc'
-                    }).then(function (data) {
-                        console.debug('TPB search: %s results', data.length);
-                        data.forEach(function (item) {
-                            if (!item.category) {
-                                return;
-                            }
-
-                            var itemModel = {
-                                title: item.name,
-                                magnet: item.magnetLink,
-                                seeds: item.seeders,
-                                peers: item.leechers,
-                                size: item.size,
-                                index: index
-                            };
-
-                            if (item.name.match(/trailer/i) !== null && item.name.match(/trailer/i) === null) {
-                                return;
+                            resolve(results1);
+                        }).catch(function (err) {
+                            console.error('rarbg search:', err);
+                            if (!retry) {
+                                return resolve(rarbg(true));
```

Comment:
why is rargb the only one with retries?

Reply:
Again.. not my code but probably because rarbg also requires a token to search and also has a 1 request / 2seconds limit. I guess the retry helps?